### PR TITLE
Introduce Docusaurus

### PR DESCRIPTION
**Pull request by 0x6b**
As a first step of tasks at issue #434, I have setup Docusaurus. Give this PR a try please, and give us your comment/advice if any. Thank you.

```sh
$ cd website
$ yarn build
$ yarn start
$ open "http://localhost:3000" # on macOS
```

Note that I haven't look into CI build failure on Node.js v4 at this moment. Will find a way around later.

```
lerna ERR! /home/travis/build/0x6b/textlint/node_modules/docusaurus/lib/server/translate-plugin.js:11
lerna ERR!   const {types: t} = babel;
lerna ERR!         ^
lerna ERR! 
lerna ERR! SyntaxError: Unexpected token {
```

Please also note that since this is WIP pull request, I will add commits and/or rebase on latest master at any time.

### CODE REVIEW

**Comment by azu** [comment]
Probably, Docusaurus doesn't support Node.js v4
Because, Node.js v4 does not support [Destructuring](http://node.green/#ES2015-syntax-destructuring--declarations)._

Revisit https://github.com/textlint/textlint/pull/442#discussion_r160368675
I think that separate jobs for JavaScript Task and Document Taks.

jobs:

- Node.js v4
  - Lint JS
  - Test
- Node.js stable
  - Lint JS
  - Test
- Document(Use Node.js stable)
  - Lint docs
  - Build Docusaurus
  - Deploy Docusaurus

**Comment by 0x6b** [comment]
Rebased on current master. Conflicting `yarn.lock` is difficult to resolve...

**Comment by 0x6b** [comment]
@azu, please review the PR as I updated this with your comments.

- Integrate [new playground](https://textlint.github.io/playground?embed)
- Separate JS and document build tasks on Travis CI

Since I'm new to React, the website does not seems to do a React way. But it can be used as basis going forward. Thanks!

File: website/sidebars.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "docs": {
+    "Docusaurus": ["doc1"],
```
Comment (azu) [inline website/sidebars.json]: We will update sidebars.json. (after merge this PR?)

[Adding Documents to a Sidebar](https://docusaurus.io/docs/en/navigation.html#adding-documents-to-a-sidebar "Adding Documents to a Sidebar")
Current limitation: [Support sub-directories in docs/ · Issue #323 · facebook/Docusaurus](https://github.com/facebook/Docusaurus/issues/323 "Support sub-directories in docs/ · Issue #323 · facebook/Docusaurus")

Comment (0x6b) [reply website/sidebars.json]: Ah, I completely missed side bars. We can improve details as you said. Thank you for reviewing 😄

**Pull request merged**